SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -198,33 +200,72 @@ private Task ProcessPendingErrorsAsync()
             });
         }
 
-        private Task SendCrashReportsOrAwaitUserConfirmationAsync()
+        private async Task SendCrashReportsOrAwaitUserConfirmationAsync()
         {
-            return HandleUserConfirmationAsync();
+            bool alwaysSend = ApplicationSettings.GetValue(PrefKeyAlwaysSend, false);
+            if (_unprocessedManagedErrorLogs.Count() > 0)
+            {
+                // Check for always send: this bypasses user confirmation callback.
+                if (alwaysSend)
+                {
+                    AppCenterLog.Debug(LogTag, "The flag for user confirmation is set to AlwaysSend, will send logs.");
+                    await HandleUserConfirmationAsync(UserConfirmation.Send);
+                    return;
+                }
+
+                var shouldAwaitUserConfirmation = ShouldAwaitUserConfirmation?.Invoke();
+                if (shouldAwaitUserConfirmation == true)
+                {
+                    AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned true, wait sending logs.");
+                }
+                else
+                {
+                    AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned false or is not implemented, will send logs.");
+                    await HandleUserConfirmationAsync(UserConfirmation.Send);
+                }
+            }
         }
 
-        private Task HandleUserConfirmationAsync()
+        private Task HandleUserConfirmationAsync(UserConfirmation userConfirmation)
         {
-            // Send every pending log.
             var keys = _unprocessedManagedErrorLogs.Keys.ToList();
             var tasks = new List<Task>();
-            foreach (var key in keys)
+
+            if (userConfirmation == UserConfirmation.DontSend)
             {
-                var log = _unprocessedManagedErrorLogs[key];
-                tasks.Add(Channel.EnqueueAsync(log));
-                _unprocessedManagedErrorLogs.Remove(key);
-                ErrorLogHelper.RemoveStoredErrorLogFile(key);
-                var errorReport = new ErrorReport(log, null);
-
-                // This must never called while a lock is held.
-                var attachments = GetErrorAttachments?.Invoke(errorReport);
-                if (attachments == null)
+                foreach (var key in keys)
                 {
-                    AppCenterLog.Debug(LogTag, $"Crashes.GetErrorAttachments returned null; no additional information will be attached to log: {log.Id}.");
+                    _unprocessedManagedErrorLogs.Remove(key);
+                    ErrorLogHelper.RemoveStoredErrorLogFile(key);
+                    // TODO: Remove exception files
```

Comment:
Is this to be done in this PR or later?

Reply:
@dhei yes, it is part of Jae and Guillaume's PR